rgw/sfs: Object Lock - Legal holds

giubacc opened the pull request:
Object Lock - Legal holds

Extended test-sfs-object-locking.py to cover Legal Holds.
Tested Legal Hold put/removal and related object deletion activity.

Fixes: aquarist-labs/s3gw#355
Signed-off-by: Giuseppe Baccini <giuseppe.baccini@suse.com>

<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

jecluis commented on src/rgw/driver/sfs/object.cc:
```diff
@@ -319,7 +319,7 @@ int SFSObject::set_obj_attrs(const DoutPrefixProvider *dpp,
                              optional_yield y) {
   ceph_assert(objref);
-  auto meta = objref->meta;
+  auto &meta = objref->meta;
```
these changes seem part of a different logical commit than the one described by this commit?

giubacc replied:
done

jecluis reviewed (changes requested):
shouldn't this be two distinct logical commits? One being the test, the other about fixing obtaining an object by reference instead of copying it?

giubacc commented:
> shouldn't this be two distinct logical commits? One being the test, the other about fixing obtaining an object by reference instead of copying it?

Those fixes were introduced because these specific tests failed, so I've incorporated all into one commit.
But yes, I can split this in two. :+1:

jecluis commented:
Thanks. I think it then makes sense. Having the first commit of the series being the bug fixes, and the second the test enhancement, would be great IMO. :)

Commit:
rgw/sfs: fix uneffective edit of attributes in object.cc

Signed-off-by: Giuseppe Baccini <giuseppe.baccini@suse.com>

Commit:
rgw/sfs: Object Lock - Legal holds

Extended test-sfs-object-locking.py to cover Legal Holds.
Tested Legal Hold put/removal and related object deletion activity.

Fixes: aquarist-labs/s3gw#355
Signed-off-by: Giuseppe Baccini <giuseppe.baccini@suse.com>

giubacc commented:
@jecluis @irq0 can we merge this?
